core/src/main/java/org/apache/calcite/sql/validate/ListScope.java

```diff
@@ -221,9 +232,15 @@ public RelDataType resolveColumn(String columnName, SqlNode ctx) {
     case 1:
       return type;
     default:
+      // Only return type if unknown tables are allowed, and at most one of the matched tables
+      // is a known table (if more than one known table is matched, then the column would be
+      // ambiguous even with full schema information). If more than one unknown table is matched,
+      // it doesn't matter which one the column actually comes from, since the column type is ANY.
+      if (validator.config().allowUnknownTables() && found - unknownFound <= 1) {
```

Comment:
Wouldn't this match if, say, found == 2 and unknownFound == 2? Do we want that to match?

Reply:
I think we would - that would mean that both of the matched tables are unknown, which is fine. The only case we want to throw an error is when more than one known table is matched.

Reply:
SGTM